pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -172,14 +174,365 @@ public static String javaPrimitiveToObjectType(String primitiveFieldType) {
+					}
+					""").replace("$fieldName", f.nameCamelFirstLower());
+				}
+				else if (f.parent() == null) { // process sub message
+				    generatedCodeSoFar += (
+               		"""
+					if ($fieldName != null && !$fieldName.equals(DEFAULT.$fieldName)) {
+					    result = 31 * result + $fieldName.hashCode();
+					}
+					""").replace("$fieldName", f.nameCamelFirstLower());
+				}
+				else {
+					throw new RuntimeException("Unexpected field type for getting HashCode - " + f.type().toString());
+				}
+			}
+		}
+		return generatedCodeSoFar.indent(DEFAULT_INDENT * 2);
+	}
+
+	/**
+	 * Get the hashcode codegen for a optional field.
+	 * @param generatedCodeSoFar The string that the codegen is generated into.
+	 * @param f The field for which to generate the hash code.
+	 * @return Updated codegen string.
+	 */
+	@NonNull
+	private static String getOptionalHashCodeGeneration(String generatedCodeSoFar, Field f) {
+		switch (f.messageType()) {
+			case "StringValue" -> generatedCodeSoFar += (
+     			"""
+				if ($fieldName != null && !$fieldName.equals(DEFAULT.$fieldName)) {
+				    result = 31 * result + $fieldName.hashCode();
+				}
+				""").replace("$fieldName", f.nameCamelFirstLower());
+			case "BoolValue" -> generatedCodeSoFar += (
+				"""
+				if ($fieldName != null && !$fieldName.equals(DEFAULT.$fieldName)) {
+				    result = 31 * result + Boolean.hashCode($fieldName);
+				}
+				""").replace("$fieldName", f.nameCamelFirstLower());
+			case "Int32Value", "UInt32Value" -> generatedCodeSoFar += (
+     			"""
+				if ($fieldName != null && !$fieldName.equals(DEFAULT.$fieldName)) {
+				    result = 31 * result + Integer.hashCode($fieldName);
+				}
+				""").replace("$fieldName", f.nameCamelFirstLower());
+			case "Int64Value", "UInt64Value" -> generatedCodeSoFar += (
+     			"""
+				if ($fieldName != null && !$fieldName.equals(DEFAULT.$fieldName)) {
+				    result = 31 * result + Long.hashCode($fieldName);
+				}
+				""").replace("$fieldName", f.nameCamelFirstLower());
+			case "FloatValue" -> generatedCodeSoFar += (
+     			"""
+				if ($fieldName != null && !$fieldName.equals(DEFAULT.$fieldName)) {
+				    result = 31 * result + Float.hashCode($fieldName);
+				}
+				""").replace("$fieldName", f.nameCamelFirstLower());
+			case "DoubleValue" -> generatedCodeSoFar += (
+     			"""
+				if ($fieldName != null && !$fieldName.equals(DEFAULT.$fieldName)) {
+				    result = 31 * result + Double.hashCode($fieldName);
+				}
+				""").replace("$fieldName", f.nameCamelFirstLower());
+			case "BytesValue" -> generatedCodeSoFar += (
+     			"""
+				if ($fieldName != null && !$fieldName.equals(DEFAULT.$fieldName)) {
+				    result = 31 * result + ($fieldName == null ? 0 : $fieldName.hashCode());
+				}
+				""").replace("$fieldName", f.nameCamelFirstLower());
+			default -> throw new UnsupportedOperationException("Unhandled optional message type:" + f.messageType());
```

Comment:
What about other messages? I mean, if field type is another PBJ message class. An example is accountId field (of type AccountID) in the Account class.

Reply:
This particular method is designed for handling specific set of classes. Take a look at `SingleField#optionalValueType` definition:
```
	public boolean optionalValueType() { // Move logic for checking built in types to common
		return type == SingleField.FieldType.MESSAGE && (
				messageType.equals("StringValue") ||
				messageType.equals("Int32Value") ||
				messageType.equals("UInt32Value") ||
				messageType.equals("Int64Value") ||
				messageType.equals("UInt64Value") ||
				messageType.equals("FloatValue") ||
				messageType.equals("DoubleValue") ||
				messageType.equals("BoolValue") ||
				messageType.equals("BytesValue")
		);
	}
```
As you can see, it includes only primitive value wrappers and doesn't include any value class in general. Therefore, this condition 
```
} else if (f.optionalValueType()) {
    generatedCodeSoFar = getOptionalHashCodeGeneration(generatedCodeSoFar, f);
} 
```
is not going to be true.